src/MassTransit.Newtonsoft/Serialization/JsonConverters/InterfaceProxyConverter.cs

```diff
@@ -16,7 +16,8 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 
         protected override IConverter ValueFactory(Type objectType)
         {
-            if (objectType.GetTypeInfo().IsInterface && MessageTypeCache.IsValidMessageType(objectType))
+            if (objectType.GetTypeInfo().IsInterface && MessageTypeCache.IsValidMessageType(objectType)
+                && objectType.GetCustomAttribute<ExcludeFromProxyGenerationAttribute>() == null)
```

Comment:
`.HasAttribute<>()`

Reply:
That extension method hasn't really been optimized, but yeah. It should be since it's currently using the non-generic version.